src/main/java/com/example/springbootapp/service/EmployeeService.java

```diff
@@ -24,23 +26,30 @@ public EmployeeService(EmployeeRepository employeeRepository, EmployeeMapper emp
         this.employeeMapper = employeeMapper;
     }
 
-    public List<EmployeeDTO> showAllEmployees() {
+    public List<EmployeeDTO> getAllEmployees() {
 
         return employeeRepository.findAll().stream().map(employeeMapper::employeeToEmployeeDTO).collect(Collectors.toList());
     }
 
-    public EmployeeDTO showEmployeeById(long id) {
+    public EmployeeDTO getEmployeeById(long id) {
         Optional<Employee> foundEmployee = employeeRepository.findById(id);
-        return employeeMapper.employeeToEmployeeDTO(foundEmployee.get());
+        return employeeMapper.employeeToEmployeeDTO(foundEmployee.orElseThrow(()-> new EmployeeNotFoundException("Employee with id " + id + " doesn't exist")));
     }
+
     @Transactional
     public Employee saveNewEmployee(EmployeeDTO employeeDTO) {
+        if (employeeDTO.isAtLeastOneParamNull(employeeDTO)){
+            throw new EmployeeNotSavedException("Employee not saved because at least one field is null");
+        }
         return employeeRepository.save(employeeMapper.employeeDTOtoEmployee(employeeDTO));
     }
 
     @Transactional
     public Employee updateEmployee(long employeeId, EmployeeDTO updatedEmployeeDTO) {
-        Employee employee = employeeRepository.findById(employeeId).get();
+        Employee employee = employeeRepository.findById(employeeId).orElseThrow(()-> new EmployeeNotFoundException("Employee with id " + employeeId + " doesn't exist"));
+        if (updatedEmployeeDTO.isAtLeastOneParamNull(updatedEmployeeDTO)){
```

Comment:
Could you do this validation before calling DB? It reduces the amount of unnecessary requests into this DB.